docs/tips/14-communicate-between-components.md

```diff
@@ -13,23 +13,21 @@ For child-parent communication:
 Say your `GroceryList` component has a list of items generated through an array. When a list item is clicked, you want to display its name:
 
 ```js
-var GroceryList = React.createClass({
-  handleClick: function(i) {
-    console.log('You clicked: ' + this.props.items[i]);
-  },
-
-  render: function() {
-    return (
-      <div>
-        {this.props.items.map(function(item, i) {
-          return (
-            <div onClick={this.handleClick.bind(this, i)} key={i}>{item}</div>
-          );
-        }, this)}
-      </div>
-    );
+function GroceryList(props) {
+  var handleClick = function(i) {
```

Comment:
We can move this up out of the GroceryList component/function, and have it accept both props and index.  That saves us an allocation on each render, which reduces GC pressure, and is therefore probably a better practice.